### PR TITLE
Lifecycle fix: gettable -> get & settable -> set

### PR DESCRIPTION
**Pull request by paulidale**
Many of the lifecycle calls were gettable/settable rather than get/set.

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man7/life_cycle-kdf.pod [hunk 1]
```diff
@@ -81,8 +81,8 @@ This is the canonical list.
  EVP_KDF_CTX_reset                           newed        newed
  EVP_KDF_CTX_get_params                      newed       deriving
  EVP_KDF_CTX_set_params                      newed       deriving
- EVP_KDF_CTX_gettable_params                 newed       deriving
- EVP_KDF_CTX_settable_params                 newed       deriving
+ EVP_KDF_CTX_get_params                      newed       deriving
```
Comment (t8m) [inline doc/man7/life_cycle-kdf.pod:84]: IMO this change is incorrect - you have get/set_params above already. The original was IMO correct.

File: doc/man7/life_cycle-mac.pod [hunk 2]
```diff
@@ -99,10 +99,10 @@ This is the canonical list.
...
  EVP_MAC_final                                                  finaled
  EVP_MAC_CTX_free                freed   freed       freed       freed       freed
- EVP_MAC_gettable_ctx_params             newed    initialised   updated    
- EVP_MAC_settable_ctx_params             newed    initialised   updated
- EVP_MAC_CTX_gettable_params             newed    initialised   updated
- EVP_MAC_CTX_settable_params             newed    initialised   updated
+ EVP_MAC_get_ctx_params                  newed    initialised   updated    
+ EVP_MAC_set_ctx_params                  newed    initialised   updated
+ EVP_MAC_CTX_get_params                  newed    initialised   updated
+ EVP_MAC_CTX_set_params                  newed    initialised   updated
```
Comment (t8m) [inline doc/man7/life_cycle-mac.pod:105]: This does not look correct either. The original is wrong but the new is wrong either. There should be these 4 functions EVP_MAC_CTX_get/set_params and EVP_MAC_CTX_gettable/settable_params.

File: doc/man7/life_cycle-rand.pod [hunk 3]
```diff
@@ -94,8 +94,8 @@ This is the canonical list.
  EVP_RAND_CTX_free          freed   freed      freed          freed
  EVP_RAND_get_ctx_params            newed   instantiated  uninstantiated  freed
  EVP_RAND_set_ctx_params            newed   instantiated  uninstantiated  freed
- EVP_RAND_CTX_gettable_params       newed   instantiated  uninstantiated  freed
- EVP_RAND_CTX_settable_params       newed   instantiated  uninstantiated  freed
+ EVP_RAND_CTX_get_params            newed   instantiated  uninstantiated  freed
+ EVP_RAND_CTX_set_params            newed   instantiated  uninstantiated  freed
```
Comment (t8m) [inline doc/man7/life_cycle-rand.pod:98]: The original was correct. And yeah - it is awkward as the EVP_RAND_get/set_ctx_params() does not follow the naming pattern from EVP_MAC and EVP_KDF. But that rename is for a different PR.

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by paulidale** [comment]
I screwed things up pretty badly.  Should be fixed now.  Only the CTX functions are listed so, e.g., `EVP_MAC_settable_ctx_params` isn't listed since that's not an operation on the CTX.

File: doc/man7/life_cycle-mac.pod [hunk 4]
```diff
@@ -153,21 +153,21 @@ This is the canonical list.
...
-<tr><th style="border:1px solid" align="left">EVP_MAC_settable_ctx_params</th>
+<tr><th style="border:1px solid" align="left">EVP_MAC_CTX_set_params</th>
     <td style="border:1px solid" align="center"></td>
     <td style="border:1px solid" align="center">newed</td>
     <td style="border:1px solid" align="center">initialised</td>
     <td style="border:1px solid" align="center">updated</td>
     <td style="border:1px solid" align="center"></td>
     <td style="border:1px solid" align="center"></td></tr>
-<tr><th style="border:1px solid" align="left">EVP_MAC_CTX_gettable_params</th>
+<tr><th style="border:1px solid" align="left">EVP_MAC_CTX_getablet_params</th>
```
Comment (slontis) [inline doc/man7/life_cycle-mac.pod:170]: tablet?

Comment (paulidale) [reply doc/man7/life_cycle-mac.pod:170]: That was easy to fix :)

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master, thanks for the comments.

**Pull request closed without merging**